### DISCUSSION TITLE
5.0（白杨） Roadmap

### DISCUSSION THREAD

**Discussion by awmleer** [ant-design/ant-design-mobile#3924] [Announcements]
5.0（白杨） 是我们最新在开发的下一代 antd-mobile 组件库，经过近 5 个月的开发，已经覆盖了 48 个组件，并已在许多项目中落地使用。

我们最近发布了 alpha 版本的 [npm 包](https://www.npmjs.com/package/antd-mobile)，也部署了新的[文档站点](https://next.mobile.ant.design)。坦白地讲，antd mobile 5.0 还并不够成熟，但我们希望能够尽早地推送给社区上的各位同学，也希望整个研发进程和规划尽可能地符合开源精神：透明、开放、合作。

所以，为了帮助大家更好地了解 5.0 版本，这里简单分享一下我们的思路和方向。

# 5.0 会带来什么

### 视觉规范

和 v3 v4 版本一致，v5 也将沿用最新版本的支付宝基础设计规范 [Alipay Design](https://design.alipay.com)。

### 手势和动画
v5 使用了 [use-gesture](https://github.com/pmndrs/use-gesture) 作为手势库、[react-spring](https://github.com/pmndrs/react-spring) 作为动画库，具有更流畅细腻的手势交互和动画效果。

### 重新设计的 API

v5 所有的组件都是完全重写的，API 也是重新设计的，更现代化也更优雅。

### 拥抱 css 变量

css 变量提供了更加动态化的样式调整能力，也让组件的样式调整变得更加简洁优雅。在业务中对组件样式魔改是一件非常痛苦也非常难以维护的事情，我们希望通过 css 变量改变这一现状。

# 了解更多

如果你想了解如何使用，可以[去这里](https://mobile.ant.design/guide/quick-start)

如果你想点点试试各种组件，可以[去这里](https://mobile.ant.design/components/button)

此外，[在这里](https://mobile.ant.design/guide/faq)我们汇总了一些常见问题，希望能帮到你

# 发布计划

> 这些里程碑只是我们大致的计划，随着项目的不断推进，可能会有所调整

### Alpha

我们在 7 月 12 日发布了第一个 alpha 版本，经过了一个多月的不断迭代，API 渐渐趋于稳定。

### Beta

我们在 8 月底开始推送了 beta 版本，相较于 alpha 版本，beta 版本 API 更加稳定也更加完善。

在这个阶段我们还会增加更多的新组件：

- [x] Swiper
- [x] ImageUpload
- [x] Stepper
- [x] SwipeAction
- [x] Navbar
- [x] NumberKeyboard
- [x] VirtualInput
- [x] Sidebar
- [x] SafeArea
- [x] PickerView 系列
- [x] FloatingBubble
- [x] AutoCenter

补充组件库的整体能力：

- [x] 支持国际化
- [x] 暴露出更多的 css 变量
- [x] 支持 SSR
- [x] 增加英文文档

### RC

我们在 11 月底开始推送了 rc 版本。
在这期间我们将几乎不会再引入新的 break change。

- [ ] 为核心组件增加自动化测试
- [ ] 完善英文文档 `WIP`

还有一些组件也可能会在 RC 阶段被新增：

- [ ] Skeleton
- [ ] SegmentedControl
- [ ] Calendar

### Release

我们预计在 2022 年 1 月正式发布 5.0 版本。

### 更远

在正式发布之后，我们可能还会在这些方向继续探索，这些新特性或许会在未来的大版本中引入，目前还没有准确的规划：

- [ ] 不断完善自动化测试
- [ ] 支持无障碍
- [ ] 支持暗黑模式

---

最后，感谢近期在内部版本的 v5 中参与建设的各位同学：

@awmleer
@Coco-rq
@ttys026
@brickspert
@zxcchen
@Deturium
@slr724038822
@quqiuhua
@crazylxr
@mhc568710904
@ustczy
@TaoBufan

也感谢几位 Sponsor 的大力支持：

@brickspert
@chenkan
@afc163

**Comment by CaicoLeung** [comment]
沙发

**Reply by Hanshaoshuai** [reply]
安装使用了css-loader编译好像不兼容var(--adm-color-white)无法编译

**Reply by Hanshaoshuai** [reply]
![image](https://user-images.githubusercontent.com/25164862/134711931-b83ed894-6a38-4ff2-b2f9-11bacbf6945a.png)

**Reply by Hanshaoshuai** [reply]
可以推荐一下那个版本loader兼容

**Reply by awmleer** [reply]
`var(--adm-color-white)` 是 CSS 变量，不需要编译
BTW 你回复到别人的评论下面了……建议单独提个 issue 吧

**Comment by jkiss** [comment]
期待

**Comment by kaoding** [comment]
期待11月份到来。

**Comment by fashen007** [comment]
期待

**Comment by thomas-void0** [comment]
cool

**Comment by chenkan** [comment]
期待 +10086

**Comment by afc163** [comment]
期待！卡晨老师 yyds！

**Comment by crazylxr** [comment]
来了 来了，他终于来了

**Comment by A-quarius** [comment]
卡神v5

**Comment by zhanghesong** [comment]
期待加油

**Comment by brickspert** [comment]
坚持就是胜利✌️

**Comment by xiaosenzhang** [comment]
呦呦呦！🐂🍺坏了

**Comment by CJY0208** [comment]
666

**Comment by Xuhao** [comment]
期待！希望tab组件能更健全！

**Comment by Coco-rq** [comment]
荣幸之至，祝star UP UP UP!

**Comment by zhaohuanlei** [comment]
终于要来了

**Comment by dengshicheng1996** [comment]
day day up

**Comment by Me-UserName** [comment]
加油，日历组件希望早点出来

**Comment by xiexingen** [comment]
点赞+收藏

**Reply by snakeUni** [reply]
cool

**Comment by neverlous** [comment]
只能说期待了~

**Comment by pigbayspy** [comment]
非常感谢！

**Comment by silverWolf818** [comment]
cool ！！！

**Comment by xieyezi** [comment]
期待已久！！

**Comment by gmars** [comment]
等不及了 先用为敬

**Comment by DreamPWJ** [comment]
期待正式版

**Comment by hzhren** [comment]
antd永远的神，V5将补齐mobile最后的短板，期待呀

**Comment by aushion** [comment]
来吧，冲我来

**Comment by zq12580** [comment]
先用为快

**Comment by danhua1** [comment]
期待10月正式发版。

**Comment by zk8080** [comment]
期待正式版！

**Comment by codedart2018** [comment]
还需要backtop

**Comment by Mrzhx** [comment]
冲啊 兄dei们...

**Comment by snakeUni** [comment]
对应的分支是 v5 分支么

**Reply by awmleer** [reply]
现在已经是 master 分支了

**Reply by snakeUni** [reply]
请教个问题，我在用 `SwipeAction` 的时候会配个上下滚动，看起来上下滚动的时候即使是手势不是垂直的也不会影响 `SwipeAction` 的横向。这个是怎么做的呀。通常上下滚动如果是斜着的会影响到 `SwipeAction` 的横向才对。

**Reply by awmleer** [reply]
https://use-gesture.netlify.app/docs/options/#axis

**Comment by gwsbhqt** [comment]
@awmleer  有计划适配微信/支付宝小程序吗？如果能够适配小程序的话适用性可能会更广

**Reply by awmleer** [reply]
目前没有要适配小程序的计划

**Reply by xiexingen** [reply]
> 目前没有要适配小程序的计划

比如 taro3

**Reply by awmleer** [reply]
目前也没有适配 taro3 的计划

**Reply by GOWxx** [reply]
考虑在小程序里用 webview 访问 用 ant-design-mobie 做的 h5 页面？

**Reply by awmleer** [reply]
@GOWxx 的确可以这么玩，不过要是这样的话，何必还用小程序……

**Reply by GOWxx** [reply]
@awmleer 考虑这种方式的主要原因是客户想从微信小程序作为入口，免了手动输入账户信息；次要原因是已有一部分钉钉小程序，现在要做一些新的图表数据展示在微信小程序上，做成H5方便两边都能用。
但似乎微信小程序只用作壳和鉴权的话，不一定过审，我这次试试。
感谢回复 ^ ^

**Comment by dingyiming** [comment]
看来暗黑模式 得等2022了

**Comment by velison** [comment]
迫不及待了，已准备更换最新版

**Comment by kaoding** [comment]
RC 了

**Comment by brave-tomato** [comment]
666🎉🎉🎉

**Comment by marchiartur** [comment]
awesome!!

**Comment by wr404274848** [comment]
Table组件有计划吗，能异步设置DataSource的

**Reply by awmleer** [reply]
Table 相关的在另一个讨论里： #4016

**Comment by zhouengineer** [comment]
regeneratorruntime is not undefined 引用version5后报这个错 没有兼容处理babel吗

**Reply by awmleer** [reply]
单独提一个 issue 哈

**Comment by kaoding** [comment]
为什么又延期了？

**Reply by awmleer** [reply]
和小程序组件库协调到一起正式发布了，预计 2 月底前发布

**Reply by codedart2018** [reply]
> 

小程序是什么计划呀
没听说过呀

**Reply by awmleer** [reply]
是兄弟团队做的支付宝小程序组件库，和 antd-mobile follow 的是同一套设计规范

**Comment by xiexingen** [comment]
支持微信小程序嘛？

**Reply by awmleer** [reply]
不支持的
